File: 260419/자동차 단일 거래 이익 최대화하기/max-profit-of-single-car.py
# n년
n = int(input())

# n년간 자동차 가격
car_price = list(map(int, input().split()))

min_price = car_price[0]
max_revenue = 0

for i in range (1, n):
    revenue = car_price[i] - min_price

    if revenue > max_revenue:
        max_revenue = revenue

    if car_price[i] < min_price:
        min_price = car_price[i]

print(max_revenue)

File: 260419/자동차 단일 거래 이익 최대화하기/max-profit-of-single-car.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        // Please write your code here.
        Scanner sc = new Scanner(System.in);

        // n 년
        int n = sc.nextInt();

        // n 년간 자동차 가격
        int[] carPrice = new int[n];

        for (int i = 0; i < n; i++) {
            carPrice[i] = sc.nextInt();
        }
        
        int minPrice = carPrice[0];
        int maxRevenue = 0;

        for (int i = 1; i < n; i++) {
            int revenue = carPrice[i] - minPrice;
            
            if (revenue > maxRevenue) {
                maxRevenue = revenue;
            }

            if (carPrice[i] < minPrice) {
                minPrice = carPrice[i];
            }
        }
        System.out.print(maxRevenue);
        sc.close();
    }
}
